### PR TITLE
fixes #5553: part 1: FiltersHelper#search_path returns an empty string for unrecognized resources

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

**Comment by ehelms** [comment]
ACK from Katello side, tested locally and was able to load our engine permissions in the UI and select them.

**Comment by dmitri-d** [comment]
[test]

File: app/helpers/filters_helper.rb [hunk 1]
```diff
@@ -11,11 +11,16 @@ def search_path(type)
         when 'Parameter'
           '' # parameter is only used in API
         else
-          resource_path(type) + '/auto_complete_search'
+          resource_path = resource_path(type)
+          resource_path.nil? ? "" : resource_path + auto_complete_search_path
```
Comment (domcleal) [inline app/helpers/filters_helper.rb:15]: Nitpick, should this be `.blank?` rather than nil? as resource_path can return ''

Comment (dmitri-d) [inline app/helpers/filters_helper.rb:15]: I could replace that with .blank?, but this is overly defensive code in 'resource_path' method -- the method never gets called if type is nil: https://github.com/witlessbird/foreman/blob/5553-part1/app/helpers/filters_helper.rb#L4

Comment (domcleal) [inline app/helpers/filters_helper.rb:15]: Ah yes, thanks.

**Comment by domcleal** [comment]
Merged as 0c82622 for Foreman 1.5.1, thanks @witlessbird!

I filed http://projects.theforeman.org/issues/5753 for the customisation of autocomplete_path.

**Pull request closed without merging**